File: JSConfig/Transactions.js
$(document).ready(function () {

 
    function calculateTotal() {

        var cashSales = parseFloat($('#TxtMembershipFee').val()) || 0;
        var shareCapital = parseFloat($('#TxtShares').val()) || 0;
        var timeDeposits = parseFloat($('#TxtTimeDeposits').val()) || 0;
        var total = cashSales + shareCapital + timeDeposits;
        $('#TxtTotalRegistration_').val(total.toFixed(2)); // Display total with two decimals
    }

    function validateShares() {
        var subscriptionAmount = parseFloat($('#TxtSubsription').val()) || 0;
        var sharesAmount = parseFloat($('#TxtShares').val()) || 0;

        if (sharesAmount > subscriptionAmount) {
            alert("Share Capital Subscription must be less than or equal to Subscription Amount.");
            $('#TxtShares').val(subscriptionAmount.toFixed(2)); // Set shares to subscription amount if invalid
        }
    }

    $('input[type="text"]').on('input', function () {
        this.value = this.value.replace(/[^0-9.]/g, ''); // Allow only numbers and decimals
        calculateTotal();
        validateShares();
    });
    calculateTotal();
});


$('#ProcessTransaction').click(function (e) {
    e.preventDefault(); // Prevents the default action
  
    var MembersFeeData = {
        EntryBy: $('#EntryBy').text(),
        MemberNo: $('#TxtNo_').text().trim(),
        Total: $('#TxtTotalRegistration_').val().trim(),
        MembershipFee: $('#TxtMembershipFee').val(),
        Subscription: $('#TxtSubsription').val(),
        Shares: $('#TxtShares').val(),
        TimeDeposits: $('#TxtTimeDeposits').val()
     
        // Exclude hidden fields
    };

    $.ajax({
        url: '/Transactions/PostTransaction',
        type: 'POST',
        data: MembersFeeData,
        success: function (response) {
            alert("Successfully Posted");
            window.location.href = '/Members/Profile_?No_=' + $('#TxtNo_').text().trim();

            // Handle success
        },
        error: function (xhr, status, error) {
            // Handle error
        }
    });
});

File: DataLibrary/Models/Transactions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataLibrary.Models
{
    public class Transactions
    {
        public int Timestamp { get; set; }
        public DateTime EntryDate { get; set; }
        public string EntryBy { get; set; }

        public string MemberNo { get; set; }

        public string MemberName { get; set; }
        public string No_ { get; set; }
        public string Type { get; set; }
        public string Month { get; set; }
        public string Year { get; set; }
        public string others { get; set; }

        public double othersFee { get; set; }
        public double Loan { get; set; }
        public double MembershipFee { get; set; }
        public double Subscription { get; set; }
        public double Shares { get; set; }
        public double TimeDeposits { get; set; }

        public double Total { get; set; }
        public string Status { get; set; }

        public string Posted { get; set; }
    }
}
